Bcrypt fails to build on io.js 2.0.0

Bcrypt fails to build when using io.js 2.0.0 on Heroku. Here are the logs, some of them may be related to other libraries: 

```
remote: Building source:
remote: 
remote: -----> Node.js app detected
remote: 
remote: -----> Reading application state
remote:        package.json...
remote:        build directory...
remote:        cache directory...
remote:        environment variables...
remote: 
remote:        ACHIEVEMENT UNLOCKED: iojs :)
remote: 
remote:        Node engine:         stable (iojs)
remote:        Npm engine:          unspecified
remote:        Start mechanism:     Procfile
remote:        node_modules source: package.json
remote:        node_modules cached: true
remote: 
remote:        NPM_CONFIG_PRODUCTION=true
remote:        NODE_MODULES_CACHE=true
remote: 
remote: -----> Installing binaries
remote:        Resolving iojs version stable via semver.io...
remote:        Downloading and installing iojs 2.0.0...
remote:        Using default npm version: 2.9.0
remote: 
remote: -----> Building dependencies
remote:        Installing node modules
remote:        
remote:        > bcrypt@0.8.2 install /tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/bcrypt
remote:        > node-gyp rebuild
remote:        
remote:        make: Entering directory `/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/bcrypt/build'
remote:          CXX(target) Release/obj.target/bcrypt_lib/src/blowfish.o
remote:          CXX(target) Release/obj.target/bcrypt_lib/src/bcrypt.o
remote:          CXX(target) Release/obj.target/bcrypt_lib/src/bcrypt_node.o
remote:        In file included from ../node_modules/nan/nan_new.h:190:0,
remote:                         from ../node_modules/nan/nan.h:80,
remote:                         from ../src/bcrypt_node.cc:1:
remote:        ../node_modules/nan/nan_implementation_12_inl.h: In static member function ‘static NanIntern::FactoryBase<v8::Signature>::return_t NanIntern::Factory<v8::Signature>::New(NanIntern::Factory<v8::Signature>::FTH, int, NanIntern::Factory<v8::Signature>::FTH*)’:
remote:        ../node_modules/nan/nan_implementation_12_inl.h:181:76: error: no matching function for call to ‘v8::Signature::New(v8::Isolate*, NanIntern::Factory<v8::Signature>::FTH&, int&, NanIntern::Factory<v8::Signature>::FTH*&)’
remote:           return v8::Signature::New(v8::Isolate::GetCurrent(), receiver, argc, argv);
remote:                                                                                    ^
remote:        ../node_modules/nan/nan_implementation_12_inl.h:181:76: note: candidate is:
remote:        In file included from /app/.node-gyp/2.0.0/src/node.h:42:0,
remote:                         from ../node_modules/nan/nan.h:24,
remote:                         from ../src/bcrypt_node.cc:1:
remote:        /app/.node-gyp/2.0.0/deps/v8/include/v8.h:4188:27: note: static v8::Local<v8::Signature> v8::Signature::New(v8::Isolate*, v8::Handle<v8::FunctionTemplate>)
remote:           static Local<Signature> New(
remote:                                   ^
remote:        /app/.node-gyp/2.0.0/deps/v8/include/v8.h:4188:27: note:   candidate expects 2 arguments, 4 provided
remote:        ../src/bcrypt_node.cc: In member function ‘virtual void {anonymous}::SaltAsyncWorker::HandleOKCallback()’:
remote:        ../src/bcrypt_node.cc:85:59: warning: ‘v8::Local<v8::Value> node::Encode(const void*, size_t, node::encoding)’ is deprecated (declared at /app/.node-gyp/2.0.0/src/node.h:278): Use Encode(isolate, ...) [-Wdeprecated-declarations]
remote:                 argv[1] = Encode(salt.c_str(), salt.size(), BINARY);
remote:                                                                   ^
remote:        In file included from ../src/bcrypt_node.cc:1:0:
remote:        ../src/bcrypt_node.cc: In function ‘void {anonymous}::GenerateSaltSync(const v8::FunctionCallbackInfo<v8::Value>&)’:
remote:        ../src/bcrypt_node.cc:139:53: warning: ‘v8::Local<v8::Value> node::Encode(const void*, size_t, node::encoding)’ is deprecated (declared at /app/.node-gyp/2.0.0/src/node.h:278): Use Encode(isolate, ...) [-Wdeprecated-declarations]
remote:             NanReturnValue(Encode(salt, strlen(salt), BINARY));
remote:                                                             ^
remote:        ../node_modules/nan/nan.h:306:65: note: in definition of macro ‘NanReturnValue’
remote:         # define NanReturnValue(value) return args.GetReturnValue().Set(value)
remote:                                                                         ^
remote:        ../src/bcrypt_node.cc: In member function ‘virtual void {anonymous}::EncryptAsyncWorker::HandleOKCallback()’:
remote:        ../src/bcrypt_node.cc:172:67: warning: ‘v8::Local<v8::Value> node::Encode(const void*, size_t, node::encoding)’ is deprecated (declared at /app/.node-gyp/2.0.0/src/node.h:278): Use Encode(isolate, ...) [-Wdeprecated-declarations]
remote:                     argv[1] = Encode(output.c_str(), output.size(), BINARY);
remote:                                                                           ^
remote:        In file included from ../src/bcrypt_node.cc:1:0:
remote:        ../src/bcrypt_node.cc: In function ‘void {anonymous}::EncryptSync(const v8::FunctionCallbackInfo<v8::Value>&)’:
remote:        ../src/bcrypt_node.cc:223:61: warning: ‘v8::Local<v8::Value> node::Encode(const void*, size_t, node::encoding)’ is deprecated (declared at /app/.node-gyp/2.0.0/src/node.h:278): Use Encode(isolate, ...) [-Wdeprecated-declarations]
remote:             NanReturnValue(Encode(bcrypted, strlen(bcrypted), BINARY));
remote:                                                                     ^
remote:        ../node_modules/nan/nan.h:306:65: note: in definition of macro ‘NanReturnValue’
remote:         # define NanReturnValue(value) return args.GetReturnValue().Set(value)
remote:                                                                         ^
remote:        make: *** [Release/obj.target/bcrypt_lib/src/bcrypt_node.o] Error 1
remote:        make: Leaving directory `/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/bcrypt/build'
remote:        gyp ERR! build error 
remote:        gyp ERR! stack Error: `make` failed with exit code: 2
remote:        gyp ERR! stack     at ChildProcess.onExit (/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/.heroku/node/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
remote:        gyp ERR! stack     at emitTwo (events.js:87:13)
remote:        gyp ERR! stack     at ChildProcess.emit (events.js:169:7)
remote:        gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1009:12)
remote:        gyp ERR! System Linux 3.13.0-49-generic
remote:        gyp ERR! command "/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/.heroku/node/bin/iojs" "/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/.heroku/node/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
remote:        gyp ERR! cwd /tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/bcrypt
remote:        gyp ERR! node -v v2.0.0
remote:        gyp ERR! node-gyp -v v1.0.3
remote:        gyp ERR! not ok 
remote:        npm WARN installMany nopt was bundled with node-pre-gyp@0.6.4, but bundled package wasn't found in unpacked tree
remote:        npm WARN installMany npmlog was bundled with node-pre-gyp@0.6.4, but bundled package wasn't found in unpacked tree
remote:        npm WARN installMany request was bundled with node-pre-gyp@0.6.4, but bundled package wasn't found in unpacked tree
remote:        npm WARN installMany semver was bundled with node-pre-gyp@0.6.4, but bundled package wasn't found in unpacked tree
remote:        npm WARN installMany tar was bundled with node-pre-gyp@0.6.4, but bundled package wasn't found in unpacked tree
remote:        npm WARN installMany tar-pack was bundled with node-pre-gyp@0.6.4, but bundled package wasn't found in unpacked tree
remote:        npm WARN installMany mkdirp was bundled with node-pre-gyp@0.6.4, but bundled package wasn't found in unpacked tree
remote:        npm WARN installMany rc was bundled with node-pre-gyp@0.6.4, but bundled package wasn't found in unpacked tree
remote:        npm WARN installMany rimraf was bundled with node-pre-gyp@0.6.4, but bundled package wasn't found in unpacked tree
remote:        
remote:        > kerberos@0.0.10 install /tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/kerberos
remote:        > (node-gyp rebuild 2> builderror.log) || (exit 0)
remote:        
remote:        make: Entering directory `/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/kerberos/build'
remote:          CXX(target) Release/obj.target/kerberos/lib/kerberos.o
remote:        make: Leaving directory `/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/kerberos/build'
remote:        
remote:        > bson-ext@0.1.3 install /tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/bson/node_modules/bson-ext
remote:        > node-pre-gyp install --fallback-to-build
remote:        
remote:        make: Entering directory `/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/bson/node_modules/bson-ext/build'
remote:          CXX(target) Release/obj.target/bson/ext/bson.o
remote:        In file included from ../node_modules/nan/nan_new.h:190:0,
remote:                         from ../node_modules/nan/nan.h:80,
remote:                         from ../ext/bson.h:17,
remote:                         from ../ext/bson.cc:36:
remote:        ../node_modules/nan/nan_implementation_12_inl.h: In static member function ‘static NanIntern::FactoryBase<v8::Signature>::return_t NanIntern::Factory<v8::Signature>::New(NanIntern::Factory<v8::Signature>::FTH, int, NanIntern::Factory<v8::Signature>::FTH*)’:
remote:        ../node_modules/nan/nan_implementation_12_inl.h:181:76: error: no matching function for call to ‘v8::Signature::New(v8::Isolate*, NanIntern::Factory<v8::Signature>::FTH&, int&, NanIntern::Factory<v8::Signature>::FTH*&)’
remote:           return v8::Signature::New(v8::Isolate::GetCurrent(), receiver, argc, argv);
remote:                                                                                    ^
remote:        ../node_modules/nan/nan_implementation_12_inl.h:181:76: note: candidate is:
remote:        In file included from ../ext/bson.cc:14:0:
remote:        /app/.node-gyp/2.0.0/deps/v8/include/v8.h:4188:27: note: static v8::Local<v8::Signature> v8::Signature::New(v8::Isolate*, v8::Handle<v8::FunctionTemplate>)
remote:           static Local<Signature> New(
remote:                                   ^
remote:        /app/.node-gyp/2.0.0/deps/v8/include/v8.h:4188:27: note:   candidate expects 2 arguments, 4 provided
remote:        ../ext/bson.cc: In static member function ‘static void BSON::BSONDeserialize(const v8::FunctionCallbackInfo<v8::Value>&)’:
remote:        ../ext/bson.cc:825:44: warning: ‘ssize_t node::DecodeBytes(v8::Handle<v8::Value>, node::encoding)’ is deprecated (declared at /app/.node-gyp/2.0.0/src/node.h:297): Use DecodeBytes(isolate, ...) [-Wdeprecated-declarations]
remote:           ssize_t len = DecodeBytes(args[0], BINARY);
remote:                                                    ^
remote:        ../ext/bson.cc:833:41: warning: ‘ssize_t node::DecodeWrite(char*, size_t, v8::Handle<v8::Value>, node::encoding)’ is deprecated (declared at /app/.node-gyp/2.0.0/src/node.h:310): Use DecodeWrite(isolate, ...) [-Wdeprecated-declarations]
remote:           DecodeWrite(data, len, args[0], BINARY);
remote:                                                 ^
remote:        ../ext/bson.cc: In static member function ‘static void BSON::BSONSerialize(const v8::FunctionCallbackInfo<v8::Value>&)’:
remote:        ../ext/bson.cc:928:73: warning: ‘v8::Local<v8::Value> node::Encode(const void*, size_t, node::encoding)’ is deprecated (declared at /app/.node-gyp/2.0.0/src/node.h:278): Use Encode(isolate, ...) [-Wdeprecated-declarations]
remote:           Local<Value> bin_value = Encode(serialized_object, object_size, BINARY)->ToString();
remote:                                                                                 ^
remote:        make: *** [Release/obj.target/bson/ext/bson.o] Error 1
remote:        make: Leaving directory `/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/bson/node_modules/bson-ext/build'
remote:        gyp ERR! build error 
remote:        gyp ERR! stack Error: `make` failed with exit code: 2
remote:        gyp ERR! stack     at ChildProcess.onExit (/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/.heroku/node/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
remote:        gyp ERR! stack     at emitTwo (events.js:87:13)
remote:        gyp ERR! stack     at ChildProcess.emit (events.js:169:7)
remote:        gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1009:12)
remote:        gyp ERR! System Linux 3.13.0-49-generic
remote:        gyp ERR! command "/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/.heroku/node/bin/iojs" "/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/.heroku/node/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/bson/node_modules/bson-ext/build/Release/bson.node" "--module_name=bson" "--module_path=/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/bson/node_modules/bson-ext/build/Release"
remote:        gyp ERR! cwd /tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/bson/node_modules/bson-ext
remote:        gyp ERR! node -v v2.0.0
remote:        gyp ERR! node-gyp -v v1.0.3
remote:        gyp ERR! not ok 
remote:        node-pre-gyp ERR! build error 
remote:        node-pre-gyp ERR! stack Error: Failed to execute '/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/.heroku/node/bin/iojs /tmp/build_ec0307398c67a7231b6dfd32b2835fa1/.heroku/node/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/bson/node_modules/bson-ext/build/Release/bson.node --module_name=bson --module_path=/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/bson/node_modules/bson-ext/build/Release' (1)
remote:        node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/bson/node_modules/bson-ext/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
remote:        node-pre-gyp ERR! stack     at emitTwo (events.js:87:13)
remote:        node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:169:7)
remote:        node-pre-gyp ERR! stack     at maybeClose (child_process.js:953:16)
remote:        node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1020:5)
remote:        node-pre-gyp ERR! System Linux 3.13.0-49-generic
remote:        node-pre-gyp ERR! command "/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/.heroku/node/bin/iojs" "/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/bson/node_modules/bson-ext/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
remote:        node-pre-gyp ERR! cwd /tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/bson/node_modules/bson-ext
remote:        node-pre-gyp ERR! node -v v2.0.0
remote:        node-pre-gyp ERR! node-pre-gyp -v v0.6.4
remote:        node-pre-gyp ERR! not ok 
remote:        Failed to execute '/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/.heroku/node/bin/iojs /tmp/build_ec0307398c67a7231b6dfd32b2835fa1/.heroku/node/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/bson/node_modules/bson-ext/build/Release/bson.node --module_name=bson --module_path=/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/bson/node_modules/bson-ext/build/Release' (1)
remote:        npm WARN optional dep failed, continuing bson-ext@0.1.3
remote:        
remote:        > bson@0.3.1 install /tmp/build_ec0307398c67a7231b6dfd32b2835fa1/node_modules/mongodb/node_modules/mongodb-core/node_modules/bson
remote:        > (node-gyp rebuild 2> builderror.log) || (exit 0)
remote:        
remote:        npm ERR! Linux 3.13.0-49-generic
remote:        npm ERR! argv "/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/.heroku/node/bin/iojs" "/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/.heroku/node/bin/npm" "install" "--unsafe-perm" "--quiet" "--userconfig" "/tmp/build_ec0307398c67a7231b6dfd32b2835fa1/.npmrc"
remote:        npm ERR! node v2.0.0
remote:        npm ERR! npm  v2.9.0
remote:        npm ERR! code ELIFECYCLE
remote:        
remote:        npm ERR! bcrypt@0.8.2 install: `node-gyp rebuild`
remote:        npm ERR! Exit status 1
remote:        npm ERR! 
remote:        npm ERR! Failed at the bcrypt@0.8.2 install script 'node-gyp rebuild'.
remote:        npm ERR! This is most likely a problem with the bcrypt package,
remote:        npm ERR! not with npm itself.
remote:        npm ERR! Tell the author that this fails on your system:
remote:        npm ERR!     node-gyp rebuild
remote:        npm ERR! You can get their info via:
remote:        npm ERR!     npm owner ls bcrypt
remote:        npm ERR! There is likely additional logging output above.
remote:        
remote:        npm ERR! Please include the following file with any support request:
remote:        npm ERR!     /tmp/build_ec0307398c67a7231b6dfd32b2835fa1/npm-debug.log
remote: 
remote: -----> Build failed
remote: 
remote:        We're sorry this build is failing!
remote:        
remote:        Are you running into a common issue?
remote:        https://devcenter.heroku.com/articles/troubleshooting-node-deploys
remote:        
remote:        If you're stuck, please submit a ticket so we can help:
remote:        https://help.heroku.com/
remote:        
remote:        Love,
remote:        Heroku
remote: 
remote:  !     Push rejected, failed to compile Node.js app
```

me too :(
```
npm i -g node-gyp && node-gyp clean                          :(
/.nvm/versions/io.js/v2.0.0/bin/node-gyp -> /.nvm/versions/io.js/v2.0.0/lib/node_modules/node-gyp/bin/node-gyp.js
node-gyp@1.0.3 /.nvm/versions/io.js/v2.0.0/lib/node_modules/node-gyp
├── which@1.0.9
├── rimraf@2.3.3
├── osenv@0.1.0
├── graceful-fs@3.0.6
├── semver@4.3.4
├── fstream@1.0.4 (inherits@2.0.1)
├── nopt@3.0.1 (abbrev@1.0.5)
├── minimatch@1.0.0 (sigmund@1.0.0, lru-cache@2.6.2)
├── mkdirp@0.5.0 (minimist@0.0.8)
├── tar@1.0.3 (inherits@2.0.1, block-stream@0.0.7)
├── glob@4.5.3 (inherits@2.0.1, inflight@1.0.4, once@1.3.2, minimatch@2.0.7)
├── npmlog@1.2.0 (ansi@0.3.0, are-we-there-yet@1.0.4, gauge@1.2.0)
└── request@2.55.0 (caseless@0.9.0, json-stringify-safe@5.0.0, aws-sign2@0.5.0, forever-agent@0.6.1, stringstream@0.0.4, oauth-sign@0.6.0, tunnel-agent@0.4.0, isstream@0.1.2, node-uuid@1.4.3, qs@2.4.1, bl@0.9.4, combined-stream@0.0.7, form-data@0.2.0, mime-types@2.0.10, http-signature@0.10.1, tough-cookie@1.1.0, hawk@2.3.1, har-validator@1.7.0)
gyp info it worked if it ends with ok
gyp info using node-gyp@1.0.3
gyp info using node@2.0.0 | linux | x64
gyp info ok 
```
```
npm install bcrypt --save          
-
> bcrypt@0.8.2 install /v2/node_modules/bcrypt
> node-gyp rebuild

make: Entering directory '/v2/node_modules/bcrypt/build'
  CXX(target) Release/obj.target/bcrypt_lib/src/blowfish.o
  CXX(target) Release/obj.target/bcrypt_lib/src/bcrypt.o
  CXX(target) Release/obj.target/bcrypt_lib/src/bcrypt_node.o
In file included from ../node_modules/nan/nan_new.h:190:0,
                 from ../node_modules/nan/nan.h:80,
                 from ../src/bcrypt_node.cc:1:
../node_modules/nan/nan_implementation_12_inl.h: In static member function ‘static NanIntern::FactoryBase<v8::Signature>::return_t NanIntern::Factory<v8::Signature>::New(NanIntern::Factory<v8::Signature>::FTH, int, NanIntern::Factory<v8::Signature>::FTH*)’:
../node_modules/nan/nan_implementation_12_inl.h:181:76: error: no matching function for call to ‘v8::Signature::New(v8::Isolate*, NanIntern::Factory<v8::Signature>::FTH&, int&, NanIntern::Factory<v8::Signature>::FTH*&)’
   return v8::Signature::New(v8::Isolate::GetCurrent(), receiver, argc, argv);
                                                                            ^
../node_modules/nan/nan_implementation_12_inl.h:181:76: note: candidate is:
In file included from /.node-gyp/2.0.0/src/node.h:42:0,
                 from ../node_modules/nan/nan.h:24,
                 from ../src/bcrypt_node.cc:1:
/.node-gyp/2.0.0/deps/v8/include/v8.h:4188:27: note: static v8::Local<v8::Signature> v8::Signature::New(v8::Isolate*, v8::Handle<v8::FunctionTemplate>)
   static Local<Signature> New(
                           ^
/.node-gyp/2.0.0/deps/v8/include/v8.h:4188:27: note:   candidate expects 2 arguments, 4 provided
../src/bcrypt_node.cc: In member function ‘virtual void {anonymous}::SaltAsyncWorker::HandleOKCallback()’:
../src/bcrypt_node.cc:85:59: warning: ‘v8::Local<v8::Value> node::Encode(const void*, size_t, node::encoding)’ is deprecated (declared at /.node-gyp/2.0.0/src/node.h:278): Use Encode(isolate, ...) [-Wdeprecated-declarations]
         argv[1] = Encode(salt.c_str(), salt.size(), BINARY);
                                                           ^
In file included from ../src/bcrypt_node.cc:1:0:
../src/bcrypt_node.cc: In function ‘void {anonymous}::GenerateSaltSync(const v8::FunctionCallbackInfo<v8::Value>&)’:
../src/bcrypt_node.cc:139:53: warning: ‘v8::Local<v8::Value> node::Encode(const void*, size_t, node::encoding)’ is deprecated (declared at /.node-gyp/2.0.0/src/node.h:278): Use Encode(isolate, ...) [-Wdeprecated-declarations]
     NanReturnValue(Encode(salt, strlen(salt), BINARY));
                                                     ^
../node_modules/nan/nan.h:306:65: note: in definition of macro ‘NanReturnValue’
 # define NanReturnValue(value) return args.GetReturnValue().Set(value)
                                                                 ^
../src/bcrypt_node.cc: In member function ‘virtual void {anonymous}::EncryptAsyncWorker::HandleOKCallback()’:
../src/bcrypt_node.cc:172:67: warning: ‘v8::Local<v8::Value> node::Encode(const void*, size_t, node::encoding)’ is deprecated (declared at /.node-gyp/2.0.0/src/node.h:278): Use Encode(isolate, ...) [-Wdeprecated-declarations]
             argv[1] = Encode(output.c_str(), output.size(), BINARY);
                                                                   ^
In file included from ../src/bcrypt_node.cc:1:0:
../src/bcrypt_node.cc: In function ‘void {anonymous}::EncryptSync(const v8::FunctionCallbackInfo<v8::Value>&)’:
../src/bcrypt_node.cc:223:61: warning: ‘v8::Local<v8::Value> node::Encode(const void*, size_t, node::encoding)’ is deprecated (declared at /.node-gyp/2.0.0/src/node.h:278): Use Encode(isolate, ...) [-Wdeprecated-declarations]
     NanReturnValue(Encode(bcrypted, strlen(bcrypted), BINARY));
                                                             ^
../node_modules/nan/nan.h:306:65: note: in definition of macro ‘NanReturnValue’
 # define NanReturnValue(value) return args.GetReturnValue().Set(value)
                                                                 ^
bcrypt_lib.target.mk:88: recipe for target 'Release/obj.target/bcrypt_lib/src/bcrypt_node.o' failed
make: *** [Release/obj.target/bcrypt_lib/src/bcrypt_node.o] Error 1
make: Leaving directory '/v2/node_modules/bcrypt/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/.nvm/versions/io.js/v2.0.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:169:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1009:12)
gyp ERR! System Linux 3.19.0-16-generic
gyp ERR! command "/.nvm/versions/io.js/v2.0.0/bin/iojs" "/.nvm/versions/io.js/v2.0.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /v2/node_modules/bcrypt
gyp ERR! node -v v2.0.0
gyp ERR! node-gyp -v v1.0.3
gyp ERR! not ok 
npm ERR! Linux 3.19.0-16-generic
npm ERR! argv "/.nvm/versions/io.js/v2.0.0/bin/iojs" "/.nvm/versions/io.js/v2.0.0/bin/npm" "install" "bcrypt" "--save"
npm ERR! node v2.0.0
npm ERR! npm  v2.9.0
npm ERR! code ELIFECYCLE

npm ERR! bcrypt@0.8.2 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the bcrypt@0.8.2 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the bcrypt package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls bcrypt
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /v2/npm-debug.log
```

--- a/package.json
+++ b/package.json
@@ -32,8 +32,8 @@
     "test": "node-gyp configure build && nodeunit test"
   },
   "dependencies": {
-    "bindings": "1.2.0",
-    "nan": "1.7.0"
+    "bindings": "1.2.1",
+    "nan": "1.8.4"
   },
   "devDependencies": {
     "nodeunit": ">=0.6.4"

